photutils/psf/daogroup.py

```diff
@@ -0,0 +1,129 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Functions for implementing the DAOGROUP algorithm proposed by Stetson 1987.
+"""
+
+from __future__ import division
+import numpy as np
+from astropy.table import Column, Table, vstack
+
+__all__ = ['daogroup']
+
+def daogroup(starlist, crit_separation):
+    """
+    This is an implementation of the DAOGROUP algorithm presented by
+    Stetson (1987).
+
+    DAOGROUP divides an entire starlist into sets of distinct, self-contained
+    groups of mutually overlapping stars.
+
+    GROUP accepts as input a list of stars and determines which stars are
+    close enough to be capable of adversely influencing each others' profile
+    fits.
+
+    Parameters
+    ----------
+    starlist : `~astropy.table.Table` or array-like
+        List of stars positions.
+        If `~astropy.table.Table`, columns should be named 'x_0' and 'y_0'.
+        TODO: If array-like, it should be either (x_0, y_0).
+    crit_separation : float or int
+        Distance, in units of pixels, such that any two stars separated by
+        less than this distance will be placed in the same group.
+
+    Returns
+    -------
+    group_starlist : list of `~astropy.table.Table`
```

Comment:
I would prefer to just add a column "group_id" to the existing table.
Adding columns is easy and the table has a good grouping interface: https://astropy.readthedocs.io/en/latest/table/operations.html#grouped-operations

Returning a list of tables is very specific and can really only be used for this one purpose, while the table can be prited for the paper, plotted in x,y ot see if it works, ...
If I have a list of table, I'll always need an extra for loop.